polyfactory/factories/base.py

```diff
@@ -642,18 +651,27 @@ def get_field_value(  # noqa: C901, PLR0911, PLR0912
             return cls.get_constrained_field_value(annotation=unwrapped_annotation, field_meta=field_meta)
 
         if BaseFactory.is_factory_type(annotation=unwrapped_annotation):
+            if not field_build_parameters and unwrapped_annotation in build_context["seen_models"]:
+                return None
```

Comment:
I'm not entirely sure on this as a default here. It is possible to have non-nullable recursive models by use of references or unions

Reply:
Yeah, I think this would fail if in the test you've written instead of `child: Optional[Node]` it becomes `child: Union[Node, SomeOtherNode]` right?

> use of references

Also, what did you mean by this?

Reply:
Makes sense. I've update to favour other potential values where possible and just omit from kwargs entirely.

> Also, what did you mean by this?

There are some cases where it is valid to type recursive models without an escape hatch via None or Union type. These are often in ORM. I think omitted here is the right choice and not super clear to me what is a general correct behaviour. For example,

```
@dataclass
class N:
    child: N
```
is valid a python typing but constructions would require extra functionality via having a ref then setting the child. I've added this to test cases.